src/website/apps/api/v2/viewsets/event.py

```
@@ -88,54 +87,77 @@ def get_serializer_class(self):  # type: ignore[override]
+            qs = qs.order_by('-start_date', 'order')
+
+        # Filter out soft-deleted items
+        if hasattr(Event, 'is_deleted'):
+            qs = qs.filter(is_deleted=False)
 
         logger.debug(
-            f"EventViewSet.get_queryset: optimized with select_related={select_related_fields}, prefetch_related={existing_prefetch}")
+            f"EventViewSet.get_queryset: optimized for action={action}")
         return qs
 
     def list(self, request, *args, **kwargs):
```

Ruff (0.13.1)

Unused method argument: `args`

(ARG002)